dataclass_wizard/abstractions.py

```diff
@@ -88,7 +88,7 @@ def list_to_json(cls: Type[W],
 
 
 @dataclass
-class AbstractParser(ABC):
+class AbstractParser(ABC, Generic[T, TT]):
```

Comment:
We  need two types, as there are some edge cases like in `UnionParser` or `PatternedDTParser`